src/puppetlabs/puppetdb/cli/import.clj

```diff
@@ -47,32 +48,34 @@
 (defn-validated process-tar-entry
   "Determine the type of an entry from the exported archive, and process it
   accordingly."
-  [^TarGzReader tar-reader :- TarGzReader
-   ^TarArchiveEntry tar-entry :- TarArchiveEntry
-   dest :- base-url-schema
-   command-versions :- {s/Any s/Any}]
+  [command-fn
+   tar-reader :- TarGzReader
+   tar-entry :- TarArchiveEntry
+   command-versions]
   (let [path (.getName tar-entry)]
     (condp re-find path
       (file-pattern "catalogs")
-      (do (println (format "Importing catalog from archive entry '%s'" path))
+      (do (log/infof "Importing catalog from archive entry '%s'" path)
           ;; NOTE: these submissions are async and we have no guarantee that they
           ;;   will succeed. We might want to add something at the end of the import
           ;;   that polls puppetdb until the command queue is empty, then does a
           ;;   query to the /nodes endpoint and shows the set difference between
           ;;   the list of nodes that we submitted and the output of that query
-          (client/submit-catalog dest
-                                 (:replace_catalog command-versions)
-                                 (archive/read-entry-content tar-reader)))
+          (command-fn :replace-catalog
+                      (:replace_catalog command-versions)
```

Comment:
I'm sure there's something nicer we could do here with the command names but I didn't want to dig into that here.